sw/device/tests/i2c_host_irq_test.c

```diff
@@ -182,6 +185,28 @@ static status_t fmt_overflow_irq(void) {
   return OK_STATUS();
 }
 
+static status_t rx_threshold_irq(void) {
+  TRY(dif_i2c_irq_set_enabled(&i2c, kDifI2cIrqRxThreshold, kDifToggleEnabled));
+  TRY(dif_i2c_set_watermarks(&i2c, /*rx_level=*/kDifI2cLevel8Byte,
+                             /*fmt_level=*/kDifI2cLevel4Byte));
+
+  irq_global_ctrl(false);
+  irq_fired = UINT32_MAX;
+  irq_global_ctrl(true);
+
+  uint8_t bytes[16];
+  const uint8_t kAddr[2] = {0x03, 0x21};
+  TRY(i2c_testutils_write(&i2c, kDeviceAddr, 2, kAddr, true));
+  // Try to read more than the watermark to trigger the IRQ.
+  TRY(i2c_testutils_read(&i2c, kDeviceAddr, sizeof(bytes), bytes,
```

Comment:
If I understand correctly, this relies on the fact that `i2c_tesutils_read` waits until the end of the transaction to read the data from the FIFO? Otherwise it could potentially pop data back from the FIFO faster than the device can fill it and never reach the watermark.

Reply:
Yes!